docs/partitions.md

```diff
@@ -0,0 +1,92 @@
+# Partitions
+
+In the original implementation of ringpop, if a cluster is split to multiple
+partitions, nodes in each partition declare each other as faulty, and don't
+communicate. Ringpop implemented support for merging the partitions, which we
+call `healing`.
+
+## Introdction -- basic algorithm
+
+In order for two partitions to heal, the algorithm does the following,
+periodically (number of times per time unit is upper bounded):
+
+1. Randomly select a `faulty` node.
+2. Send it a `/join` request, get its membership list.
+3. If the local and retrieved lists are incompatible (merging them will 
+   introduce new faulties), force all cluster to reincarnate by marking
```

Comment:
"force all cluster to reincarnate" is confusing, both "all cluster" and the verb reincarnate. I don't know what you mean to say, but reincarnate is too domain specific. Perhaps: "force the entire cluster to ...", with a different action/description.